### ISSUE TITLE
"Subprocess exiting" message and conda-build issues

### ISSUE THREAD

**Issue by msarahan** [conda/conda#3242]
The new behavior with "Subprocess exiting", from https://github.com/conda/conda/blob/e37a65c6f1ec7bf972ccb6c532f41bb5e5923e0f/conda/exceptions.py#L103-L106 seems to be causing a lot of issues with conda-build.  For example, there's the index command:

```
 msarahan@0109-msarahan-2  ~/code/conda-build   create_python_api  conda index tmp
updating index in: /Users/msarahan/code/conda-build/tmp
Subprocess exiting
 0%
```

Strange thing here is that the actual meat of the command seemed to work OK (index files created as expected), but the error code returned is failure.

```                                                                                                            
✘ msarahan@0109-msarahan-2  ~/code/conda-build   create_python_api  ls tmp
repodata.json     repodata.json.bz2
```

This command works fine with conda 4.1.11.

**Comment by HugoTian** [comment]
We can delete the error message, and conda 4.1.x, there is just sys.exit() without message.

**Comment by msarahan** [comment]
I'm not sure that the message itself is the whole issue. Some things, such as 

    Conda build --output 

Will be messed up, but AFAICT there should not be anything that sets an error code other than 0.

**Comment by HugoTian** [comment]
I already change back to sys.exit() in conda_argparse.py, and it no longer use SubprocessExit from exception. It is in PR #3245

**Comment by kalefranz** [comment]
Yup, already merged for conda 4.2.2.
